### PR TITLE
fix: handle `defer` in LIFO order

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- We have been noticing lots of `call to released function` errors lately...
- After some digging, it turns out that `preprocessStyle` was queueing `defer`'d callbacks incorrectly.
- This updates `preprocessStyle` to properly account for `defer`'s LIFO behavior.

## Testing

Tested manually, very hard to automatically test

## Docs

Bug fix only

### CODE REVIEW

**Comment by matthewp** [comment]
lgtm

File: internal_wasm/utils/utils.go [hunk 1]
```diff
@@ -9,25 +9,27 @@ import (
 // See https://stackoverflow.com/questions/68426700/how-to-wait-a-js-async-function-from-golang-wasm
 func Await(awaitable js.Value) ([]js.Value, []js.Value) {
 	then := make(chan []js.Value)
-	defer close(then)
 	thenFunc := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
 		then <- args
 		return nil
 	})
+	// defers are called LIFO!
```
Comment (drwpow) [inline internal_wasm/utils/utils.go:16]: 🙇🏻 Thank you! This is very helpful

File: internal_wasm/utils/utils.go [hunk 2]
```diff
@@ -9,25 +9,27 @@ import (
...
+	// This will `close` before `Release()`
+	defer thenFunc.Release()
```
Comment (drwpow) [inline internal_wasm/utils/utils.go:18]: Unknown, but how does the `.Call("then" …).Call("catch")` work? Do we need to wait to release `thenFunc` until `catch` has been called? Or no?

File: lib/compiler/test/basic.test.mjs [hunk 3]
```diff
@@ -43,7 +43,6 @@ div {
 `,
     {
       sourcemap: true,
-      // HOLY CRAP THIS ACTUALLY WORKS!
```
Comment (drwpow) [inline lib/compiler/test/basic.test.mjs:46]: RIP

**Review by drwpow — APPROVED** [review]
Looks good! Only have a question but this definitely seems like an improvement

File: internal_wasm/utils/utils.go (hunk 2 above)
Comment (natemoo-re) [reply internal_wasm/utils/utils.go:18]: All `defers` will be called after `return`, so they clean themselves up automatically.

**Pull request merged**